src/components/OrderDetails.js

```diff
@@ -1,14 +1,62 @@
 import { Component, PropTypes, DOM as dom, createFactory } from 'react';
+import { connect } from 'react-redux';
+import actions from '../action_creators';
+import _ from 'underscore';
+
+function mapStateToProps(state) {
+  return {
+    orders: state.orders
```

Comment:
I'm thinking the same about having order details know all orders...it's a smelly smell. How's the router stuff coming?



> On Nov 10, 2015, at 01:39, Andy Chiu <notifications@github.com> wrote:
> 
> In src/components/OrderDetails.js:
> 
> > @@ -1,14 +1,62 @@
> >  import { Component, PropTypes, DOM as dom, createFactory } from 'react';
> > +import { connect } from 'react-redux';
> > +import actions from '../action_creators';
> > +import _ from 'underscore';
> > +
> > +function mapStateToProps(state) {
> > +  return {
> > +    orders: state.orders
> So I'm not sure it's best to have a specific OrderDetails know about / have access to all orders. Ideally, we would just pull the data for the particular order (status, entries, etc.) and map those to props. But I don't think we can access the id param from state right..? That's why I was looking into redux-router. This is probably fine for now though..
> 
> —
> Reply to this email directly or view it on GitHub.
>